File: service/src/main/java/com/boxfox/cross/service/coin/CoinServiceImpl.java
package com.boxfox.cross.service.coin;

import static io.one.sys.db.Tables.COIN;

import com.linkbit.android.entity.CoinModel;
import java.util.List;
import org.jooq.DSLContext;

public class CoinServiceImpl {

  public List<CoinModel> getAllCoins(DSLContext ctx) {
    List<CoinModel> coins = ctx.selectFrom(COIN).fetch().map(coinRecord -> {
      CoinModel coinModel = new CoinModel();
      coinModel.setName(coinRecord.getName());
      coinModel.setSymbol(coinRecord.getSymbol());
      return coinModel;
    });
    return coins;
  }
}

File: service/src/main/java/com/boxfox/cross/service/coin/CoinServiceImpl.kt
package com.boxfox.cross.service.coin

import com.boxfox.cross.common.entity.CoinModel
import io.one.sys.db.Tables.COIN
import io.one.sys.db.tables.records.CoinRecord

import org.jooq.DSLContext

class CoinServiceImpl : CoinUsecase{

    override fun getAllCoins(ctx: DSLContext): List<CoinModel> {
        return ctx.selectFrom<CoinRecord>(COIN).fetch().map { coinRecord ->
            val coinModel = CoinModel()
            coinModel.name = coinRecord.getName()
            coinModel.symbol = coinRecord.getSymbol()
            coinModel
        }
    }
}
